src/api/handlers/user.ts

```diff
@@ -0,0 +1,29 @@
+import { RequestManager } from "../../rest/request_manager.ts";
+import { CreateGuildPayload, UserPayload } from "../../types/mod.ts";
+import { endpoints } from "../../util/constants.ts";
+
+/**
+ * Returns the user object of the requester's account.
+ * For OAuth2, this requires the `identify` scope, which will return the object without an email,
+ * and optionally the email scope, which returns the object with an email.
+ */
+export function getCurrentUser() {
+  return RequestManager.get(endpoints.USER_BOT) as Promise<
+    UserPayload
+  >;
+}
+
+/**
+ * Returns a list of partial guild objects the current user is a member of. 
+ * Requires the `guilds` OAuth2 scope.
+ * 
+ * ⚠️ **If you need this, you are probably doing something wrong. Always use cache.guilds
```

Comment:
I still think this comment is wrong big time. This is not about getting the bots guilds but the end user who is logging in

Reply:
Please double check this is what the function does

Reply:
https://discord.com/developers/docs/resources/user#get-current-user-guilds

Reply:
Hmm I think ima remove that